### DISCUSSION TITLE
【Help】附加新文件的快捷键

### DISCUSSION THREAD

**Discussion by felicity1559** [windingwind/zotero-actions-tags#287] [Q&A]
【Help】很多文献条目并不能自动抓取到PDF全文，手动下载好文献全文，需要右键附加新文件；是否可以设置快捷键省去右键找到再点击，感谢！

**Comment by ramonmi** [comment]
我有这个需求也好多天了，看社区里没有此类脚本，自己动手学着写了一个，欢迎测试交流、批评指正。<https://github.com/windingwind/zotero-actions-tags/discussions/290>

**Reply by felicity1559** [reply]
非常感谢，刚刚试用了下，非常不错，要是可以跟右键附加新文件一样实现自动添加最新的pdf就更好了！！！

**Reply by ramonmi** [reply]
您所说的右键添加新文件自动添加最新的PDF是否借助了插件呢，这个代码和右键Add Attachment --> File的功能应该是一样的

**Reply by felicity1559** [reply]
应该是Zotero Attanger的插件
![image](https://github.com/windingwind/zotero-actions-tags/assets/126839331/83879c16-005b-4582-84b1-a8881d1dca6d)

**Reply by ramonmi** [reply]
是的，这个插件实现了Zotero6 Zotfile的功能，目前我没有使用这个插件的需求，不过看一下这个插件的重命名api接口，应该就可以实现你的需求

**Reply by felicity1559** [reply]
嗯嗯 谢谢！对我来说太难了，我先将就着😯

**Reply by DereckZhang** [reply]
终于找到了替代zotfile的插件，从实际的操作效率来说，能不能做成copy-paste模式，桌面/资源管理器里面Ctrl+C复制了文件，zotero里选择item，按快捷键就直接添加刚刚copy的文件到条目下，如果没有选择条目，就直接作为独立文件添加到当前分类下，谢谢！

**Reply by ramonmi** [reply]
2026-5-30: https://github.com/windingwind/zotero-actions-tags/discussions/290 新增功能：自动附加指定路径的最新的PDF

**Reply by ramonmi** [reply]
> 从实际的操作效率来说，能不能做成copy-paste模式，桌面/资源管理器里面Ctrl+C复制了文件，zotero里选择item，按快捷键就直接添加刚刚copy的文件到条目下，如果没有选择条目，就直接作为独立文件添加到当前分类下

@DereckZhang implemented in #618

**Reply by windingwind** [reply]
zotero是可以读写文件的

**Reply by ramonmi** [reply]
我只找到了`Zotero.Utilities.Internal.getClipboard("text/plain")`读取剪贴板文本，剪贴板文件读取的具体命令是在哪里呢

**Reply by windingwind** [reply]
```js
function f() {
    var clip = Services.clipboard;
    var flavor = "application/x-moz-file";
    if (!clip.hasDataMatchingFlavors([flavor], 1, clip.kGlobalClipboard)) {
        return null;
    }
    var trans = Components.classes["@mozilla.org/widget/transferable;1"]
        .createInstance(Components.interfaces.nsITransferable);
    trans.init(null);
    trans.addDataFlavor(flavor);
    clip.getData(trans, clip.kGlobalClipboard);
    var data = {};
    try {
        trans.getTransferData(flavor, data, {});
        return data.value.QueryInterface(Components.interfaces.nsIFile);
    }
    catch (e) {
        return null;
    }
}
```

**Reply by ramonmi** [reply]
Thanks, @windingwind. The function for importing the clipboard file was implemented in #618.
